builtin/providers/google/provider.go

```diff
@@ -264,3 +265,63 @@ func handleNotFoundError(err error, d *schema.ResourceData, resource string) err
 
 	return fmt.Errorf("Error reading %s: %s", resource, err)
 }
+
+var regexpUrl = regexp.MustCompile(fmt.Sprintf("^(?:https://www.googleapis.com/compute/%s/)?(projects/%s(?:/(?:global|regions/%s|zones/%s)(?:/%s/%s)?)?)$",
+	"([a-z0-9]+)",                    // API Version
+	"([a-z](?:[-a-z0-9]*[a-z0-9])?)", // Project ID
+	"([a-z]+-[a-z]+[0-9]+)",          // Region
+	"([a-z]+-[a-z]+[0-9]+-[a-z]+)",   // Zone
+	"([a-z][A-Za-z]*(?:/family)?)",   // Resource type
+	"([a-z](?:[-a-z0-9]*[a-z0-9])?)", // Resource name
+))
+
+type resourceInfo struct {
+	apiVersion   string
+	project      string
+	region       string
+	zone         string
+	resourceType string
+	name         string
+	url          string
+}
+
+func parseUrl(url string) (resourceInfo, error) {
```

Comment:
So after discussing with @danawillow, I think we'd rather a more specific, less-encompassing function that just takes a URL (full or partial) for specifically the certs, and always returns the full URL. Something like [what we do for disk images](https://github.com/hashicorp/terraform/blob/a47d0544a996e0f4f7ec648ab68943ed242d258a/builtin/providers/google/image.go#L70), though hopefully more minimal. The reason we think it'd be better to do a more specific version is because our resources tend to have shorthand inputs (like `family/image` or just `image` for images) to make the UX more intuitive and easier to work with. I also worry that the regular expression is descriptive, not prescribed, and so it may not be followed for future resources, which would lead to some maintenance headaches. :( So for the time being, I think specific solutions are probably better than generic ones. But thank you for the hard work on this! It definitely lays the groundwork if we want to move in this direction in the future.